src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp

```diff
@@ -393,8 +393,7 @@ queryDispatch(Tag, Tree const &tree, ExecutionSpace const &space,
     Kokkos::Profiling::pushRegion(profiling_prefix + "::compute_permutation");
     auto permute = Details::BatchedQueries<DeviceType>::
         sortPredicatesAlongSpaceFillingCurve(space, Experimental::Morton32(),
-                                             static_cast<Box>(tree.bounds()),
-                                             predicates);
+                                             tree.bounds(), predicates);
```

Comment:
Why would you not cast to `Box<DIM>` here at the call site?

Reply:
Because I no longer require for the bounding volume to be statically casted to `Box<DIM>`. I didn't want to introduce those explicit casts to existing non-dimensional classes. Instead, I rely on `expand(Box, Geometry)` inside the function to produce the box.